### ISSUE TITLE
Unable to enable firewall on QNAP

### ISSUE THREAD

**Issue by wr0ngway** [haugene/docker-transmission-openvpn#422]
I have everything working with the firewall disabled, but when I enable the firewall (ENABLE_UFW=true), startup fails during the ufw-init due to modprobe not being found.

I tried creating a new image with modprobe installed:
```
docker run -it haugene/transmission-openvpn bash
> apt-get update && apt-get install module-init-tools
> exit
docker commit --change='CMD ["dumb-init", "/etc/openvpn/start.sh"]' <hash> haugene/transmission-openvpn
```

and while it now finds modprobe, modprobe itself fails.

Is this just not a supported option on a QNAP TVS-471 v4.3.4.0435 ?

**Comment by Mikiya83** [comment]
i cannot make it work either, i think it's related to kernel-limition with QTS. I anyone know a way i'm interested too !

**Comment by haugene** [comment]
Could it be related to #328, #499 and #559? Anyways, closing for now.
If someone still has this issue after using option from #559 and upgrading their QNAP and Docker versions, feel free to re-open.

**Comment by Mikiya83** [comment]
Unfortunatly it's not closed, I'm on the last QTS version (4.3.6.0805), with the latest docker version on QNAP (Docker version 17.09.1-ce, build 0a5102e), and i can't enable UFW.
I get logs like that : 

```
enabling firewall
ERROR: problem running ufw-init
modprobe: can't change directory to '/lib/modules': No such file or directory
modprobe: can't change directory to '/lib/modules': No such file or directory
modprobe: can't change directory to '/lib/modules': No such file or directory
iptables-restore: line 77 failed
iptables-restore: line 11 failed
sysctl: error setting key 'net.ipv4.conf.default.rp_filter': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.rp_filter': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv6.conf.default.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv6.conf.all.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv6.conf.default.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv6.conf.all.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_echo_ignore_broadcasts': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_ignore_bogus_error_responses': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_echo_ignore_all': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.log_martians': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.log_martians': Read-only file system

Problem running '/etc/ufw/before.rules'
Problem running '/etc/ufw/user.rules'

allowing 36565 through the firewall
Rules updated
allowing 10.0.3.1 through the firewall to port 9091
Rules updated
adding route to local network 192.168.1.0/24 via 10.0.3.1 dev eth0
allowing 192.168.1.0/24 through the firewall to port 9091
Rules updated
```


if i try to use option from #559 i get : 

```
sed: /usr/lib/python3/dist-packages/ufw/util.py: No such file or directory
Disable and blank firewall
Firewall stopped and disabled on system startup
enabling firewall
ERROR: problem running ufw-init
modprobe: can't change directory to '/lib/modules': No such file or directory
modprobe: can't change directory to '/lib/modules': No such file or directory
modprobe: can't change directory to '/lib/modules': No such file or directory
iptables-restore: line 77 failed
sysctl: error setting key 'net.ipv4.conf.default.rp_filter': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.rp_filter': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv6.conf.default.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv6.conf.all.accept_source_route': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv6.conf.default.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv6.conf.all.accept_redirects': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_echo_ignore_broadcasts': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_ignore_bogus_error_responses': Read-only file system
sysctl: error setting key 'net.ipv4.icmp_echo_ignore_all': Read-only file system
sysctl: error setting key 'net.ipv4.conf.default.log_martians': Read-only file system
sysctl: error setting key 'net.ipv4.conf.all.log_martians': Read-only file system

Problem running '/etc/ufw/before.rules'

allowing 36565 through the firewall
Rules updated
```

if i can try anything, tell me

**Comment by wr0ngway** [comment]
Same here, seeing the same symptoms as @Mikiya83
